test/config_test/config_test.cc

```diff
@@ -85,16 +85,37 @@ class ConfigTest {
   Runtime::RandomGeneratorImpl random_;
 };
 
+void test_merge() {
```

Comment:
Rename to `testMerge` to match Envoy style. It's a bit of an anti-pattern having these `All` pattern for tests, you could either add an independent invocation from `example_configs_test` or leave a `TODO` here to clean it up later.

Reply:
@nt that worked, no more seg faults! Can you address this comment and we can merge? Thanks.